### PR TITLE
Add area of kite and hexagon function

### PR DESCRIPTION
**Pull request by mahiuddin-dev**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/area.py [hunk 1]
```diff
@@ -458,6 +458,52 @@ def area_rhombus(diagonal_1: float, diagonal_2: float) -> float:
...
+        ...
+    ValueError: area_kite() only accepts non-negative values
+    >>> area_kite(-1, 2)
+    Traceback (most recent call last):
+        ...
+    ValueError: area_kite() only accepts non-negative values
+    """
+    if diagonal_1 < 0 or diagonal_2 < 0:
+        raise ValueError("area_kite() only accepts non-negative values")
+    return 1 / 2 * diagonal_1 * diagonal_2
```
Comment (realstealthninja) [inline maths/area.py:485]:
```suggestion
    return (diagonal_1 * diagonal_2) * 0.5
```
multiplication is always faster than division!

File: maths/area.py [hunk 2]
```diff
@@ -458,6 +458,52 @@ def area_rhombus(diagonal_1: float, diagonal_2: float) -> float:
...
+    >>> area_hexagon(0)
+    0.0
+    >>> area_hexagon(-1)
+    Traceback (most recent call last):
+        ...
+    ValueError: area_hexagon() only accepts non-negative values
+    """
+    if side < 0:
+        raise ValueError("area_hexagon() only accepts non-negative values")
+    return (3 * sqrt(3) * (side * side)) / 2
```
Comment (realstealthninja) [inline maths/area.py:504]:
```suggestion
    return (3 * sqrt(3) * (side * side))  * 0.5

```
same here multiplication is always faster than division!

File: maths/area.py [hunk 3]
```diff
@@ -458,6 +458,52 @@ def area_rhombus(diagonal_1: float, diagonal_2: float) -> float:
     return 1 / 2 * diagonal_1 * diagonal_2
 
 
+def area_kite(diagonal_1: float, diagonal_2: float) -> float:
```
Comment (realstealthninja) [inline maths/area.py:461]: @cclauss  need you opinion on something this function is quite similar to the one above like exactly the same just with different tests and a different name are similar functions allowed?

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: maths/area.py (hunk 2 above)
Comment (realstealthninja) [inline maths/area.py:504]: question: is pow(side, 2 ) or side ** 2 better than side * side?

File: maths/area.py (hunk 3 above)
Comment (cclauss) [reply maths/area.py:461]: I think that if the function above will also work for kites then we should add a comment to that function and avoid a duplicate function.

Comment (realstealthninja) [reply maths/area.py:461]: I completely agree.

File: maths/area.py (hunk 2 above)
Comment (cclauss) [reply maths/area.py]: * https://docs.python.org/3/library/timeit.html
* https://github.com/TheAlgorithms/Python/search?q=timeit

**Review by realstealthninja — CHANGES_REQUESTED** [review]

**Pull request closed without merging**